src/main/java/com/socrata/datasync/publishers/GISPublisher.java

```diff
@@ -0,0 +1,246 @@
+import java.util.logging.Logger;
+
+public class GISPublisher {
+
+    private static final Logger logging = Logger.getLogger(GISJob.class.getName());
+    private static String ticket = "";
+
+    public static JobStatus replaceGeo(File file,
+                                       SocrataConnectionInfo connectionInfo,
+                                       String datasetID,
+                                       UserPreferences userPrefs) {
+        URI scan_url = makeUri(connectionInfo.getUrl(), "scan","");
+        String blueprint = "";
+        JobStatus status = JobStatus.SUCCESS;
+        try {
+            blueprint = postRawFile(scan_url, file, userPrefs, connectionInfo);
+            status = replaceGeoFile(blueprint, file, userPrefs, connectionInfo, datasetID);
+        } catch (IOException e) {
+            String message = e.getMessage();
+            JobStatus s = JobStatus.PUBLISH_ERROR;
+            s.setMessage(message);
+            return s;
+        }
+
+        return status;
+    }
+
+    public static JobStatus replaceGeoFile(String blueprint,
+                                           File file,
+                                           UserPreferences userPrefs,
+                                           SocrataConnectionInfo connectionInfo,
+                                           String datasetID) {
+        JobStatus status = JobStatus.SUCCESS;
+        JSONParser parser = new JSONParser();
+        try {
+            Object obj = parser.parse(blueprint);
+            JSONObject array = (JSONObject)obj;
+
+            if (array.containsKey("error")) {
+                boolean error = (boolean) array.get("error");
+                if (error) {
+                    String message = array.get("message").toString();
+                    logging.log(Level.INFO,message);
+                    JobStatus s = JobStatus.PUBLISH_ERROR;
+                    s.setMessage(message);
+                    return s;
+                }
+            }
+
+            String fileId = array.get("fileId").toString();
+            String name = file.getName();
+            String bluepr = array.get("summary").toString();
+            String query = "";
+
+            query = query + "&fileId=" +  URLEncoder.encode(fileId,"UTF-8");
+            query = query + "&name=" + URLEncoder.encode(name,"UTF-8");
+            query = query + "&blueprint=" + URLEncoder.encode(bluepr,"UTF-8");
+            query = query + "&viewUid=" + URLEncoder.encode(datasetID,"UTF-8");
+
+            URI uri = makeUri(connectionInfo.getUrl(),"replace",query);
+            logging.log(Level.INFO,uri.toString());
+            status = postReplaceGeoFile(uri, connectionInfo, userPrefs);
+        } catch (ParseException | UnsupportedEncodingException e) {
+            String message = e.getMessage();
+            JobStatus s = JobStatus.PUBLISH_ERROR;
+            s.setMessage(message);
+            return s;
+        }
+
+        return status;
```

Comment:
All but one of these branches have internal returns; it'd probably be best to make the last the same,